src/VisualStudio/Core/Impl/CodeModel/CodeModelProjectCache.cs

```diff
@@ -33,9 +35,10 @@ internal sealed partial class CodeModelProjectCache
         private EnvDTE.CodeModel _rootCodeModel;
         private bool _zombied;
 
-        internal CodeModelProjectCache(IThreadingContext threadingContext, ProjectId projectId, ICodeModelInstanceFactory codeModelInstanceFactory, ProjectCodeModelFactory projectFactory, IServiceProvider serviceProvider, HostLanguageServices languageServices, VisualStudioWorkspace workspace)
+        internal CodeModelProjectCache(IThreadingContext threadingContext, IAsynchronousOperationListener listener, ProjectId projectId, ICodeModelInstanceFactory codeModelInstanceFactory, ProjectCodeModelFactory projectFactory, IServiceProvider serviceProvider, HostLanguageServices languageServices, VisualStudioWorkspace workspace)
```

Comment:
📝 ProjectCodeModelFactory has a private listener. It may simplify this PR to make that listener `internal` and use it.

Reply:
This is now simplified.